### PR TITLE
Data caching documentation

### PR DESCRIPTION
**Pull request by stagraqubole**

### CODE REVIEW

**Comment by mosabua** [comment]
This feature is currently experimental and by itself a larger feature. Personally I would prefer to have the hive caching documentation located in a separate document. We were planning to use the docs from https://docs.starburstdata.com/latest/connector/hive-cache.html and backport them to prestosql and adapt as desired. 

Maybe we should start a new PR and add some of the info from this PR in the same new doc? Or migrate some of the content from the linked docs into this PR. 

Before we do anything I think we should get input from @electrum, @dain and @martint to see what they prefer as approach.

**Comment by stagraqubole** [comment]
Sure @mosabua. We can merge the two, I opened this because there were questions around configurations in slack channels and no place for users to find the required info.

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -775,6 +775,54 @@ Now, Presto queries can take advantage of the Alluxio catalog service, such as
...
+Setting these configs for a hive catalog enables data caching on all tables in that catalog.
+Following optional configurations can be used to control the caching behavior:
+
+============================================ ================================================================== ===========
+Property Name                                Description                                                          Default
+============================================ ================================================================== ===========
+``hive.cache.parallel-warmup-enabled``       Enables asynchronous cache warmup to minimize the impact of        ``true``
+                                             cold-cache.
+
+``hive.cache.bookkeeper-port``               Defines the port to be used by the BookKeeper server.              8899
```
Comment (hashhar) [inline presto-docs/src/main/sphinx/connector/hive.rst:803]: A comment that if caching is enabled for multiple catalogs, each catalog should provide a different value for the cache location and the bookkeeper and data-transfer ports will also resolve https://github.com/prestosql/presto/issues/3723.

**Comment by mosabua** [comment]
This has been replaced with more up to date and comprehensive documentation. It is now live at https://prestosql.io/docs/current/connector/hive-caching.html .. if there are any further changes or updates you would like to propose please submit a PR with the relevant changes.

Thank you very much for getting this started @stagraqubole

**Pull request closed without merging**